mantidproject/mantid
# Initial PR for the Elemental Analysis Grouping Tab

**Description of work.**

The intitial work to develop a grouping tab for the new Elemental Analysis GUI . This includes: -

- MVP pattern for the grouping tab and table
- Description and Table column headers as agreed with the instrument scientist
- Interactive Checkbox for Analyse plot/fit column
- Drop down menu for rebin column
- Rebin options input box for steps/bins depending on Rebin selection which updates Rebin Options column
- Ability to remove a grouping
- Each workspace within each group (i.e one workspace per detector per run) added to the grouping table when a run is loaded. Functionality also works for multiple runs and co-add runs
- Grouping table is limited to 20 
- Testing for code produced in this PR

Further functionality to be added in future:-

- Rebin workspace added on based on Rebin selection and Rebin Options text


**To test:**


1. Code review
2. Before starting Mantid go to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L24 and add Muon/Elemental_Analysis_2.py to the end of the line
3. Start Mantid
4. Open the elemental analysis 2 GUI - it should appear as per screenshot below
![image](https://user-images.githubusercontent.com/55837273/102480697-91105b00-4058-11eb-92f0-64a1eda36caa.png)
5. Download the test files from PR #29958 
6. Enter 9997 in the Loading Box and press return. Click on the grouping tab and it should appear as per screenshot below
![Screenshot (68)](https://user-images.githubusercontent.com/55837273/102480981-f5cbb580-4058-11eb-8aaf-3527ab945915.png)
7. Check that you can tick and untick the Analyse Plot/Fit checkboxes
8. Check that the Rebin dropdown includes Fixed and Variable
9. Right click on a group in the table and click to remove group. The group should disappear from the table
10. Highlight several different groups, right click and remove them.

Partially Fixes #29382 

*This does not require release notes* because Grouping Tab not finished and new GUI is not being relased yet




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
